pennylane_qiskit/qiskit_device2.py

```diff
@@ -550,21 +550,37 @@ def _execute_runtime_service(self, circuits, session):
             else self.backend.configuration().backend_name
         )
 
-        options = {
+        circuit_runner_options = {
             "backend": backend_name,
             "log_level": self.options.environment.log_level,
             "job_tags": self.options.environment.job_tags,
             "max_execution_time": self.options.max_execution_time,
         }
 
+        # Does not support AerSimulator specific options e.g. choose a specific method
+        # refer to this https://qiskit.github.io/qiskit-aer/stubs/qiskit_aer.AerSimulator.html
+        # To support this would be confusing in terms of option setting (The options we currently support are runtime options)
+        # This here is just to capture and track shots information
+        aer_options = {
+            "shots": circuits[0].shots.total_shots or self.shots.total_shots,
+        }
+
         # Send circuits to the cloud for execution by the circuit-runner program.
-        job = self.service.run(
-            program_id="circuit-runner",
-            options=options,
-            inputs=program_inputs,
-            session_id=session.session_id,
-        )
-        self._current_job = job.result(decoder=RunnerResult)
+        # Cloud simulators will be deprecated on May 15th so this will be exclusively for real hardware devices.
```

Comment:
Just to confirm, have you run your hardware test notebook on this branch?

Reply:
No but I should. I'll follow up on this.

Reply:
There are some issues with running this on real hardware. Seems like last time I used the hardware I ran circuits that had operations that were supported by the hardware but this is not always the case. Per Qiskit's documentation, one needs to do the transpilation beforehand before submitting it to a real hardware device. I created a separate story to address this issue. Shouldn't be that difficult but it falls under the general "migrate everything" task, so I will probably tackle all the migration stuff at the same time.

For reference:
[sc-62928]
[Qiskit docs](https://docs.quantum.ibm.com/run/primitives-examples)

Additionally, the legacy backend.run() interface in Qiskit Runtime, which was used as the dedicated “direct hardware access” entry point, has been deprecated by Qiskit. The new SamplerV2 class now fulfills this role. Support for the backend.run() will be dropped on or around October 15, 2024. Please refer to the migration guide (https://docs.quantum.ibm.com/api/migration-guides/qiskit-runtime) for instructions on how to migrate any existing code. This corresponds to the "circuit-runner" and "qasm3-runner" programs if you are invoking the REST API directly.

Reply:
> one needs to do the transpilation beforehand before submitting it to a real hardware device

Ah. That's new. Is it using the same `compile.transpile` function that the device already uses in `_execute_runtime_service`? If so we can just move all of that up to the main execute instead.

> "the legacy backend.run() interface in Qiskit Runtime, which was used as the dedicated “direct hardware access” entry point, has been deprecated by Qiskit. The new SamplerV2 class now fulfills this role. Support for the backend.run() will be dropped on or around October 15, 2024

Is `SamplerV2` supported before Qiskit 1.0? Or would switching to that force a dependency on 1.0? Also, do you know if "dropping support" means that they don't guarantee it and won't debug it, but you can still use it if you have an old version of Qiskit installed? Or does it mean the backends won't recognize it anymore, even if you have an older version of Qiskit?

Reply:
Discussed offline.